mypy/stubtest.py

```diff
@@ -271,16 +315,16 @@ def _belongs_to_runtime(r: types.ModuleType, attr: str) -> bool:
         return not isinstance(obj, types.ModuleType)
 
     runtime_public_contents = (
-        runtime.__all__
-        if hasattr(runtime, "__all__")
-        else [
+        runtime_all_as_set | {"__all__"}
```

Comment:
I'm not sure that I like always checking for `__all__` (if the stub doesn't have `__all__` and `__all__` isn't itself in runtime's `__all__`). Either way we'll check whether we have the symbols we want, so in my mind this mostly becomes busy work.

Reply:
I'd counter that `__all__` is itself a valid variable in a Python module, though, and leaving it out of the stub can lead to false positives! 

E.g. this line in the `sphinx` codebase used to cause mypy to emit an error: https://github.com/sphinx-doc/sphinx/blob/3d6d501a43d2e5942153baa9414a4de34323c56b/sphinx/domains/python.py#L1438

See https://github.com/python/typeshed/pull/7373#issuecomment-1049430887

Reply:
But `__all__` isn't part of the public API if `__all__` doesn't contain `__all__` ;-)

Proposal:
This change has been lying around for far too long, since 90% of the change is awesome and uncontroversial, let's merge it minus the verify the presence of `__all__` bit. I'll open an issue to track verifying presence of `__all__` and I can assess whether this causes too much busywork for other users of stubtest.
```
diff --git a/mypy/stubtest.py b/mypy/stubtest.py
index 9595fa612..9a30c4be6 100644
--- a/mypy/stubtest.py
+++ b/mypy/stubtest.py
@@ -313,7 +313,7 @@ def verify_mypyfile(
         return not isinstance(obj, types.ModuleType)
 
     runtime_public_contents = (
-        runtime_all_as_set | {"__all__"}
+        runtime_all_as_set
         if runtime_all_as_set is not None
         else {
             m
```

Reply:
Alrighty, let's do it!